src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSBase.java

```diff
@@ -184,4 +184,73 @@ public void verifyDataIsTransferredToTargetGCSBucket() {
       }
     }
   }
+
+  @Then("Verify output field {string} in target BigQuery table contains filename of the source GcsBucket {string}")
+  public void verifyOutputFieldInTargetBigQueryTableContainsFilenameOfTheSourceGcsBucket(String field, String path)
+    throws IOException, InterruptedException {
+    Optional<String> result = BigQueryClient
+      .getSoleQueryResult("SELECT distinct " + PluginPropertyUtils.pluginProp(field) + " as bucket FROM `"
+                            + (PluginPropertyUtils.pluginProp("projectId")) + "."
+                            + (PluginPropertyUtils.pluginProp("dataset")) + "."
+                            + TestSetupHooks.bqTargetTable + "` ");
+    String pathFromBQTable = StringUtils.EMPTY;
+    if (result.isPresent()) {
+      pathFromBQTable = result.get();
+    }
+    BeforeActions.scenario.write("GCS bucket filename in BQ Table :" + pathFromBQTable);
+    String filename[] = PluginPropertyUtils.pluginProp(path).split("/");
+    Assert.assertEquals("BQ table output field should contain GCS bucket filename ",
+                        filename[filename.length - 1], pathFromBQTable);
+  }
+
+  @Then("Enter GCS File system properties field {string}")
+  public void enterGCSFileSystemPropertiesField(String property) {
+    CdfGcsActions.enterFileSystemProperties(PluginPropertyUtils.pluginProp(property));
+  }
+
+  @Then("Click on Tidy in GCS File system properties")
+  public void clickOnTidyInGCSFileSystemProperties() {
+    CdfGcsActions.clickTidy();
+  }
+
+  @Then("Verify data is transferred to target GCS bucket with file format {string}")
+  public void verifyDataIsTransferredToTargetGCSBucketWithFileFormat(String fileFormat) {
+    String gcsBucket = TestSetupHooks.gcsTargetBucketName;
+    if (verifyFilePresentInGcsBucket(gcsBucket, "." + fileFormat)) {
+      BeforeActions.scenario.write("Data transferred to gcs bucket " + gcsBucket + " with format "
+                                     + fileFormat + " successfully");
+    } else {
+      Assert.fail("Data not transferred to target gcs bucket " + gcsBucket + " with format " + fileFormat);
+    }
+  }
+
+  @Then("Verify data is transferred to target GCS bucket with fileName {string}")
+  public void verifyDataIsTransferredToTargetGCSBucketWithFileNamePrefix(String fileName) {
+    String gcsBucket = TestSetupHooks.gcsTargetBucketName;
+    String file = PluginPropertyUtils.pluginProp(fileName);
+    if (verifyFilePresentInGcsBucket(gcsBucket, file)) {
+      BeforeActions.scenario.write("Data transferred to gcs bucket " + gcsBucket + " with fileNameContains "
+                                     + fileName + " successfully");
+    } else {
+      Assert.fail("Data not transferred to target gcs bucket " + gcsBucket +
+                    " with fileNameContains " + fileName);
+    }
+  }
+
+  private boolean verifyFilePresentInGcsBucket(String gcsBucket, String fileNameContains) {
+    try {
```

Comment:
Add one more parameter to this method  String filePart  and 
change if condition based on filePart
if (blob.getName().contains(fileNameContains)) {
          return true;
        }

if filePart is "suffix" -> .endsWith
else if filePart is "prefix" -> .startsWith
else if filePart is "contains" -> .contains
else .equals